syntax = "proto3";

package userProfileRecommend;

// 요청 메시지: memberId를 받아서 유사한 아이템을 검색
message ProfileRequest {
  int64 memberId = 1;
  int32 page = 2;
}

// 각 아이템의 메타데이터 정보를 반환하는 메시지
message SimilarItem {
  string song_info_id = 1;        // 아이템의 ID
  string song_name = 2;           // 노래 제목
  string artist_name = 3;         // 아티스트 이름
  bool mr = 4;                    // MR 여부
  string ssss = 5;                // 추가 메타데이터 필드 (ssss)
  string audio_file_url = 6;      // 오디오 파일 URL
  float similarity_score = 7;     // 유사도 점수
}

// 유사한 아이템들의 리스트를 반환하는 응답 메시지
message ProfileResponse {
  repeated SimilarItem similar_items = 1; // 유사한 아이템 목록
}

// UserProfile 서비스
service UserProfile {
  rpc CreateUserProfile(ProfileRequest) returns (ProfileResponse) {}
}